crates/datafusion/src/lib.rs

```diff
@@ -176,6 +215,22 @@ impl TableProvider for HudiDataSource {
 
         Ok(exec_builder.build_arc())
     }
+
+    fn supports_filters_pushdown(
+        &self,
+        filters: &[&Expr],
+    ) -> Result<Vec<TableProviderFilterPushDown>> {
+        filters
+            .iter()
+            .map(|expr| {
+                if self.can_push_down(expr) {
+                    Ok(TableProviderFilterPushDown::Exact)
```

Comment:
since we only do partition pruning, we can't guarantee all filters used, so it should be `Inexact`